### PR TITLE
Docker image for the backend

### PR DESCRIPTION
**Pull request by smadbe**
* change the config depth separator for something more compatible with shells and docker-compose configs
* create a docker so that API consumer can run our backend more easily

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,23 @@ You can then run the app: (call `./bin/AlgoreaBackend` to print the list of avai
...
 
-## Running the setup
+## Running the setup (as API consumer)
+
+The easiest way to run the backend for consumer it is to run it in a container with its database. To do that:
+
+* clone this repository (or download the `docker-compose.yml` file and replace `build: .` by `image: franceioi/algoreabackend:latest` to use the public image)
+* Seed the database:
+
+  ```docker-compose run backend /bin/sh -c "sleep 1; ALGOREA_DATABASE__CONNECTION__USER="root"; ALGOREA_DATABASE__CONNECTION__PASSWD="a_root_db_password"; AlgoreaBackend db-restore; AlgoreaBackend db-migrate; AlgoreaBackend install;"```
```
Comment (zenovich) [inline README.md:27]: It doesn't work:
```
./bin/AlgoreaBackend install
2020/04/06 19:53:13 Loading environment: test
INFO 2020/04/06 19:53:13 Loading environment: test 
ERRO Error 1044: Access denied for user 'algorea'@'%' to database 'algorea_db_test'  module=database
```

Comment (zenovich) [reply README.md:27]: Actually it also requires something to be done with configs before running.

File: README.md [hunk 2]
```diff
@@ -17,7 +17,23 @@ You can then run the app: (call `./bin/AlgoreaBackend` to print the list of avai
...
+* Launch the docker compose setup (db+backend): `docker-compose up`
+* Visit `http://127.0.0.1:8080/status` with your browser, you should get a success status message.
+
+If needed, you can cannot on the MySQL CLI using:
```
Comment (zenovich) [inline README.md:31]: I can't get the phrase

File: README.md (hunk 1 above)
Comment (zenovich) [inline README.md:27]: should probably be ```docker-compose run backend /bin/sh -c "sleep 1; ALGOREA_DATABASE__CONNECTION__USER="root"; ALGOREA_DATABASE__CONNECTION__PASSWD="a_root_db_password"; AlgoreaBackend db-restore; AlgoreaBackend db-migrate; AlgoreaBackend install;"
should be
```
docker-compose run backend /bin/sh -c "sleep 1; ALGOREA_DATABASE__CONNECTION__USER="root"; ALGOREA_DATABASE__CONNECTION__PASSWD="a_root_db_password"; AlgoreaBackend db-restore && AlgoreaBackend db-migrate && AlgoreaBackend install"
``` in order to stop on errors

File: README.md (hunk 2 above)
Comment (smadbe) [reply README.md:31]: Me neither 😁 Probably "cannot" is "connect" ;-)

File: README.md (hunk 1 above)
Comment (smadbe) [reply README.md:27]: will do

File: README.md [hunk 3]
```diff
@@ -17,7 +17,23 @@ You can then run the app: (call `./bin/AlgoreaBackend` to print the list of avai
...
 
-## Running the setup
+## Running the setup (as API consumer)
+
+The easiest way to run the backend for consumer it is to run it in a container with its database. To do that:
+
+* clone this repository (or download the `docker-compose.yml` file and replace `build: .` by `image: franceioi/algoreabackend:latest` to use the public image)
+* Seed the database:
+
+  ```docker-compose run backend /bin/sh -c "sleep 1; ALGOREA_DATABASE__CONNECTION__USER="root"; ALGOREA_DATABASE__CONNECTION__PASSWD="a_root_db_password"; AlgoreaBackend db-restore && AlgoreaBackend db-migrate && AlgoreaBackend install;"```
```
Comment (zenovich) [inline README.md:27]: Do we need this semicolon at the end?

**Comment by zenovich** [comment]
Probably it would be much better to log in text format instead of json. Also, the output should be 'stdout', otherwise we get this everytime:
```
{"level":"error","msg":"Unable to open file for logs, fallback to stdout: open /go/src/AlgoreaBackend/app/logging/../../log/all.log: no such file or directory\n","time":"2020-04-07T11:38:54Z"}
```

File: README.md (hunk 3 above)
Comment (smadbe) [reply README.md]: not important,  but I removed it

File: docker-compose.yml [hunk 4]
```diff
@@ -23,18 +23,18 @@ services:
...
       ALGOREA_DATABASE__CONNECTION__ADDR: db
       ALGOREA_DATABASE__CONNECTION__USER: algorea
       ALGOREA_DATABASE__CONNECTION__PASSWD: a_db_password
       ALGOREA_DATABASE__CONNECTION__DBNAME: algorea_db
-      ALGOREA_LOGGING__FORMAT: json
-      ALGOREA_LOGGING__OUTPUT: file
-      ALGOREA_LOGGING__LOGSQLQUERIES: 1
+      ALGOREA_LOGGING__FORMAT: text
+      ALGOREA_LOGGING__OUTPUT: stdout
+      ALGOREA_LOGGING__LOGSQLQUERIES: 0
```
Comment (zenovich) [inline docker-compose.yml:33]: I still see them: 
```
[2020-04-07 12:04:24]  sql-stmt-exec map[duration:1.519171ms err:<nil> query:UPDATE groups_propagate AS children SET children.ancestors_computation_state='processing' WHERE children.ancestors_computation_state = 'todo' AND NOT EXISTS ( SELECT 1 FROM ( SELECT 1 FROM groups_groups JOIN groups_propagate ON groups_propagate.id = groups_groups.parent_group_id AND groups_propagate.ancestors_computation_state <> 'done' WHERE groups_groups.child_group_id = children.id FOR UPDATE ) has_undone_parents FOR UPDATE )]  
```

Comment (smadbe) [reply docker-compose.yml:33]: I don't :-P

**Pull request merged**